spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -155,23 +405,463 @@ class NwbSortingExtractor(se.SortingExtractor):
+        start_frame: int
+            The start frame of the epoch to be added (inclusive)
+        end_frame: int
+            The end frame of the epoch to be added (exclusive)
+        '''
+        assert HAVE_NWB, "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+        if not isinstance(epoch_name, str):
+            raise Exception("'epoch_name' must be a string")
+        if not isinstance(start_frame, int):
+            raise Exception("'start_frame' must be an integer")
+        if not isinstance(end_frame, int):
+            raise Exception("'end_frame' must be an integer")
+
+        with NWBHDF5IO(self._path, 'r+') as io:
+            nwbfile = io.read()
+            nwbfile.add_epoch(start_time=self.frame_to_time(start_frame),
+                              stop_time=self.frame_to_time(end_frame),
+                              tags=epoch_name)
+            io.write(nwbfile)
+
+    def remove_epoch(self, epoch_name=None):
+        '''NWB files do not allow removing epochs.'''
+        print(self.remove_epoch.__doc__)
+
+    def get_epoch_names(self):
+        '''This function returns a list of all the epoch names in the NWB file.
+
+        Returns
+        ----------
+        epoch_names: list
+            List of epoch names in the sorting extractor
+        '''
+        assert HAVE_NWB, "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+        with NWBHDF5IO(self._path, 'r') as io:
+            nwbfile = io.read()
+            if nwbfile.epochs is None:
+                print("No epochs in NWB file")
+                return
+            return [x[0] for x in nwbfile.epochs['tags'][:]]
+
+    def get_epoch_info(self, epoch_name):
+        '''This function returns the start frame and end frame of the epoch
+        in a dict.
+
+        Parameters
+        ----------
+        epoch_name: str
+            The name of the epoch to be returned
+
+        Returns
+        ----------
+        epoch_info: dict
+            A dict containing the start frame and end frame of the epoch
+        '''
+        assert HAVE_NWB, "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+        if not isinstance(epoch_name, str):
+            raise ValueError("epoch_name must be a string")
+        all_epoch_names = self.get_epoch_names()
+        if epoch_name not in all_epoch_names:
+            raise ValueError("This epoch has not been added")
+
+        fs = self.get_sampling_frequency()
+        epoch_info = {}
+        with NWBHDF5IO(self._path, 'r') as io:
+            nwbfile = io.read()
+            flat_list = [item for sublist in nwbfile.epochs['tags'][:] for item in sublist]
+            for i, tag in enumerate(flat_list):
+                if tag == epoch_name:
+                    epoch_info['start_frame'] = int(nwbfile.epochs['start_time'][i] * fs)
+                    epoch_info['end_frame'] = int(nwbfile.epochs['stop_time'][i] * fs)
```

Comment:
I wonder if it wouldn't be preferable to use `floor` and `ceil` used instead of pure `int` in both cases here